marti_nav_msgs/msg/Obstacle.msg

```diff
@@ -0,0 +1,10 @@
+string id
+# Unique id for the obstacle or empty if not used.
+
+geometry_msgs/PoseStamped pose
+# The pose of the origin of the obstacle.
+
+geometry_msgs/Point[] polygon
```

Comment:
I didn't use Polygon because it just adds a layer of indirection like you said, and it is built out of Point32 out of Point, making it slightly more work to convert into TF types.